tests/cnf/core/network/metallb/tests/bgp-gracefulrestart.go

```diff
@@ -127,6 +144,157 @@ var _ = Describe("BGP Graceful Restart", Ordered, Label(tsparams.LabelGRTestCase
 		})
 
 	})
+
+	Context("with bfd", func() {
+		var (
+			bfdProfile   *metallb.BFDBuilder
+			bfdConfigMap *configmap.Builder
+		)
+
+		BeforeAll(func() {
+			By("Editing the MachineConfiguration CR named cluster to change node disruption policies")
+			updateMachineConfigurationNodeDisruptionPolicy()
+		})
+
+		BeforeEach(func() {
+			By("Creating BFD profile")
+			frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+				LabelSelector: tsparams.FRRK8sDefaultLabel,
+			})
+			Expect(err).ToNot(HaveOccurred(), "Failed to list pods")
+			bfdProfile = createBFDProfileAndVerifyIfItsReady(frrk8sPods)
+
+			By("Creating MetalLb configMap")
+			bfdConfigMap = createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, true)
+		})
+		Context("single hop", func() {
+			BeforeEach(func() {
+				By("Creating BGP peer")
+				bgpPeer, err = metallb.NewBPGPeerBuilder(APIClient, "testpeer", NetConfig.MlbOperatorNamespace,
+					ipv4MetalLbIPList[0], tsparams.LocalBGPASN, tsparams.LocalBGPASN).
+					WithBFDProfile(bfdProfile.Definition.Name).WithPassword(tsparams.BGPPassword).
+					WithGracefulRestart(true).Create()
+				Expect(err).ToNot(HaveOccurred(), "Fail to create bgpPeer")
+
+				By("Creating FRR Pod with network and IP address")
+				frrPod = createFrrPod(
```

Comment:
`createFRRPod` is what go prefers, but I would change to `createExternalBGPPeer`

Reply:
This is the old function. I'll create an additional refactor PR for that purpose.